Clients/Xamarin.Interactive.Client/Telemetry/Client.cs

```diff
@@ -101,141 +129,88 @@ void ObservePreferenceChange (PreferenceChange change)
-            ITelemetryEvent evnt;
-
-            public EventObjectStreamContent (Client client, ITelemetryEvent evnt)
-            {
-                this.client = client
-                    ?? throw new ArgumentNullException (nameof (client));
-
-                this.evnt = evnt
-                    ?? throw new ArgumentNullException (nameof (evnt));
-
-                EnsureHeaders ();
-            }
-
-            void EnsureHeaders ()
-            {
-                // Mono's HttpContent seems to re-allocate 'Headers' some time after
-                // the constructor finishes, clobbering anything configured during
-                // the constructor. However, it seems to write them on the first call
-                // to Stream.Write for the Stream passed to SerializeToStreamAsync,
-                // so the fix on Mono is to configure Headers before writing to the
-                // stream. This however is not the behavior of .NET, which requires
-                // Headers to be configured in the constructor (it likely writes them
-                // explicitly to the stream before calling SerializeToStreamAsync,
-                // and not as an implementation of the Stream itself. So do both.
-                Headers.ContentType = new MediaTypeHeaderValue ("application/json");
+                appInsightsClient.TrackEvent (eventTelemetry);
+            } catch (Exception e) {
+                Log.Error (TAG, e);
+                return;
             }
 
-            protected override Task SerializeToStreamAsync (Stream stream, TransportContext context)
-            {
-                EnsureHeaders ();
-
-                var writer = new StreamWriter (stream);
-
-                try {
-                    serializer.Serialize (writer, evnt);
-                } catch (Exception e) {
-                    throw new JsonSerializationException ("bad use of JsonTextWriter / JsonSerializer", e);
-                }
+            eventsSent++;
 
-                writer.Flush ();
+            Log.Verbose (TAG, $"({eventsSent}): {eventName}");
+        }
 
-                return Task.CompletedTask;
-            }
+        public void Post (ITelemetryEvent evnt)
+        {
+            if (evnt == null)
+                throw new ArgumentNullException (nameof (evnt));
 
-            protected override bool TryComputeLength (out long length)
-            {
-                length = -1;
-                return false;
-            }
+            Post (evnt.GetType ().Name, evnt.GetProperties ());
         }
-    }
 
-    sealed class JsonSerializationException : Exception
-    {
-        public JsonSerializationException (string message, Exception innerException)
-            : base (message, innerException)
+        /// <summary>
+        /// Flush queued telemetry events synchronously.
+        /// </summary>
+        /// <param name="timeout">Defaults to 5 seconds.</param>
+        public void Flush (Optional<TimeSpan> timeout = default)
```

Comment:
One more nit, because I hate methods like this, even though it has a summary doc. Call it `BlockingFlush`?